src/engine/Match_env.ml

```diff
@@ -50,6 +50,7 @@ type xconfig = {
    *)
   matching_explanations : bool;
   filter_irrelevant_rules : bool;
+  rule_filter_cache : Analyze_rule.prefilter_cache;
```

Comment:
I'm missing where are you passing this cache to `Analyze_rule.regexp_prefilter_of_rule`. Right now it seems that we create a cache in `Match_rules.is_relevant_rule_for_xtarget`, and in `Check_rule.stat_files`, but I don't see those places being updated in this PR. Could you explain why?

Reply:
It's because I resolved some merge conflicts incorrectly while rushing to put the PR up before I had to leave 😅